frame/staking/src/tests.rs

```diff
@@ -398,8 +396,7 @@ fn less_than_needed_candidates_works() {
 		.validator_count(4)
 		.nominate(false)
 		.num_validators(3)
-		.build()
-		.execute_with(|| {
+		.build_and_execute(|| {
```

Comment:
@kianenigma ⚠️

Reply:
yeahm good point.

Reply:
to clarify, is the issue that we where missing `.execute_with(post_conditions)`?

Reply:
yes.